### PR TITLE
Initial checkin for DevicePropertiesWebHook

### PR DESCRIPTION
**Pull request by mikehall-ms**
# Description

Project that shows how to update device twin information for a device connected to Azure IoT Hub, using Event Grid and a custom ASP.NET Core application - includes ADAL for user authentication.

## Type of change

Please delete options that are not relevant.

- New content

# Checklist:

- [x] My content has a file named README.md, which is based on the appropriate readme [template](https://github.com/Azure/azure-sphere-gallery/tree/main/Templates)
- [x] I have performed a self-review of my own contribution
- [x] I have provided comments where appropriate
- [x] I have updated the repository's [README.md](https://github.com/Azure/azure-sphere-gallery/blob/main/README.md) to index this project
- [x] I have added/updated license(s) for this project as appropriate
- [x] I have permission to publish this content (in case the content was collaborative)
- [x] I have verified that my content does not contain sensitive information (PII, Microsoft PI, etc.) and is safe to open source

### CODE REVIEW

File: DevicePropertiesWebHook/README.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
+ic# Littlefs project
```
Comment (tokenstar) [inline DevicePropertiesWebHook/README.md:1]: needs update

File: DevicePropertiesWebHook/README.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+The goal of this project is to show how to update device twin properties for a device connected to Azure IoT Hub, examples might be the Product, Device Group, and OS Feed type.
```
Comment (tokenstar) [inline DevicePropertiesWebHook/README.md:3]: missing phrase "to reflect Azure Sphere Security Service (AS3) data about that device, such as its Product, ...

Comment (tokenstar) [reply DevicePropertiesWebHook/README.md:3]: Also add "The project also shows how to transmit device lifecycle events from the Azure Sphere device to Azure IoT, such as when the app is restarted or when the app receives notification that the device software is up-to-date.

**Review by cawhitworth — APPROVED** [review]
Looks good to me.

File: AzureSphereTenantDeviceTwinSync/README.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+| `src\HighLevelApp`       | Azure Sphere Sample App source files (based on the Azure IoT Github sample) |
+| `src\REST_API`       | ASP.NET Core REST API project |
+| `src\DesktopSimulator`       | C# .NET Core desktop application for simulating device telemetry |
+| `README.md` | This README file. |
+| `LICENSE.txt`   | The license for the project. |
+
+## Prerequisites & Setup
+
+- An Azure Sphere-based device with development features (see [Get started with Azure Sphere](https://azure.microsoft.com/en-us/services/azure-sphere/get-started/) for more information).
+- Setup a development environment for Azure Sphere (see [Quickstarts to set up your Azure Sphere device](https://docs.microsoft.com/en-us/azure-sphere/install/overview) for more information).
```
Comment (tokenstar) [inline AzureSphereTenantDeviceTwinSync/README.md:20]: [nit] scrub /en-us/ from docs URLs to enable automatic localization

File: AzureSphereTenantDeviceTwinSync/README.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+You will need your REST API web hook URL, this will look something like:
+
+```
+https://yourAPIServiceURL/webhook
+```
+
+Note that the APIKey is only needed to kick off the authentication process, and isn't needed for the WebHook post API.
+
+Use this [Internet of Things show](https://channel9.msdn.com/Shows/Internet-of-Things-Show/IoT-Devices-and-Event-Grid?term=event%20grid&lang-en=true) on MSDN Channel 9 as a guide for setting up IoT Hub/Event Grid with message forwarding and filtering to a WebHook (in the video the WebHook is an Azure Logic App you will point the Event Grid WebHook URL to the REST API).
```
Comment (tokenstar) [inline AzureSphereTenantDeviceTwinSync/README.md:106]: [nit] unnecessary suffix on URL

**Review by tokenstar — APPROVED** [review]

**Pull request merged**